File: SenLib/Sen2/FileFixes/scripts_scena_dat_us_e7050_dat.cs
using HyoutaUtils;
using HyoutaUtils.Streams;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SenLib.Sen2.FileFixes {
	class scripts_scena_dat_us_e7050_dat : FileMod {
		public string GetDescription() {
			return "Fix typo in Grianos' name.";
		}

		public IEnumerable<FileModResult> TryApply(FileStorage storage) {
			var file = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0xd0a9a39e3aa04d57ul, 0x3b9a7b51a170fb1dul, 0xc4d79f17u));
			if (file == null) {
				return null;
			}

			var bin = file.CopyToMemory();
			var patcher = new SenScriptPatcher(bin);

			// Girianos -> Grianos
			patcher.RemovePartialCommand(0x9369, 0xbb, 0x93de, 0x1);

			return new FileModResult[] { new FileModResult("data/scripts/scena/dat_us/e7050.dat", bin) };
		}

		public IEnumerable<FileModResult> TryRevert(FileStorage storage) {
			var file = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0xd0a9a39e3aa04d57ul, 0x3b9a7b51a170fb1dul, 0xc4d79f17u));
			if (file == null) {
				return null;
			}
			return new FileModResult[] { new FileModResult("data/scripts/scena/dat_us/e7050.dat", file) };
		}
	}
}

File: native/sen2/file_fixes/scripts_scena_dat_us_e7050_dat.cpp
#include <string_view>
#include <vector>

#include "sen/file_getter.h"
#include "p3a/pack.h"
#include "p3a/structs.h"
#include "sen/sen_script_patcher.h"
#include "util/hash/sha1.h"

namespace SenLib::Sen2::FileFixes::scripts_scena_dat_us_e7050_dat {
std::string_view GetDescription() {
    return "Fix typo in Grianos' name.";
}

bool TryApply(const SenPatcher::GetCheckedFileCallback& getCheckedFile,
              std::vector<SenPatcher::P3APackFile>& result) {
    try {
        auto file = getCheckedFile(
            "data/scripts/scena/dat_us/e7050.dat",
            126953,
            HyoutaUtils::Hash::SHA1FromHexString("d0a9a39e3aa04d573b9a7b51a170fb1dc4d79f17"));
        if (!file) {
            return false;
        }

        auto& bin = file->Data;

        SenScriptPatcher patcher(bin);

        // Girianos -> Grianos
        patcher.RemovePartialCommand(0x9369, 0xbb, 0x93de, 0x1);


        result.emplace_back(std::move(bin), file->Filename, SenPatcher::P3ACompressionType::LZ4);

        return true;
    } catch (...) {
        return false;
    }
}
} // namespace SenLib::Sen2::FileFixes::scripts_scena_dat_us_e7050_dat
